pkg/gardenlet/controller/shoot/shoot/reconciler_delete.go

```diff
@@ -239,10 +239,8 @@ func (r *Reconciler) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 			Dependencies: flow.NewTaskIDs(initializeSecretsManagement, deployCloudProviderSecret, ensureShootClusterIdentity),
 		})
 		waitUntilControlPlaneReady = g.Add(flow.Task{
-			Name: "Waiting until Shoot control plane has been reconciled",
-			Fn: flow.TaskFn(func(ctx context.Context) error {
-				return botanist.Shoot.Components.Extensions.ControlPlane.Wait(ctx)
-			}),
+			Name:         "Waiting until Shoot control plane has been reconciled",
+			Fn:           botanist.Shoot.Components.Extensions.ControlPlane.Wait,
```

Comment:
This is not correct, it will lead to nil pointer panic.

If there is a flow node that has skip condition, it can be the case that the function you are trying to call was never populated.
For ex -  in this case `ControlPlane` field is not populated for workerless, so when you run the code for workeless shoots it will panic here since you are trying to call `ControlPlane.Wait` but `ControlPlane` was never populated. [ref](https://github.com/gardener/gardener/blob/cba55148ed937263289ce52ffa839e02573bc833/pkg/gardenlet/operation/botanist/botanist.go#L81-L93)

I would recommend to carefully look all such cased before removing the `flow.TaskFn`.

Reply:
The wrapper functions were added exactly for this reason.

Reply:
Nil pointer exceptions: [ref](https://gcsweb.prow.gardener.cloud/gcs/gardener-prow/pr-logs/pull/gardener_gardener/10854/pull-gardener-e2e-kind/1857044802458095616/artifacts/gardener-local/gardener-local-control-plane/containers/gardenlet-776fc57b97-wt24p_garden_gardenlet-84885d2cedbe9fb7e4f3a4b84e0c70f532f4e45b0c1bbbbfa254bf7b0020ddfe.log)